lightblue/src/test/java/org/esbtools/eventhandler/lightblue/testing/MultiStringDocumentEvent.java

```diff
@@ -116,7 +119,8 @@ public boolean isSupersededBy(DocumentEvent event) {
 
     @Override
     public boolean couldMergeWith(DocumentEvent event) {
-        return event instanceof MultiStringDocumentEvent;
+        return event instanceof MultiStringDocumentEvent &&
+                !event.isSupersededBy(this);
```

Comment:
Yeah, but this is because you know how it is used... nothing about the contract of this type says isSuperseded should be checked before couldMergeWith. It's important actually that implementers of DocumentEvents understand that the two should be mutually exclusive.